src/main/java/seedu/address/model/slot/SlotDatePredicate.java

```diff
@@ -0,0 +1,26 @@
+package seedu.address.model.slot;
+
+import java.time.LocalDateTime;
+
+/**
+ * Tests that a {@code Slot}'s {@code DateTime} matches the given date.
+ */
+public class SlotDatePredicate extends SlotPredicate {
+    private final LocalDateTime dateTime;
+
+    public SlotDatePredicate(LocalDateTime dateTime) {
+        this.dateTime = dateTime;
+    }
+
+    @Override
+    public boolean test(Slot slot) {
+        return dateTime.toLocalDate().isEqual(slot.getDateTime().toLocalDate());
```

Comment:
can make use of [Slot#getDate](https://github.com/AY1920S2-CS2103-W15-4/main/pull/25/files#diff-d754cd73ef34866e67fe3c9a244c3d6fR47) instead